modules/vebal/voting-list.service.ts

```diff
@@ -0,0 +1,105 @@
+import { formatFixed } from '@ethersproject/bignumber';
+import { formatUnits } from 'ethers/lib/utils';
+import { chunk, zipObject } from 'lodash';
+import { networkContext } from '../../modules/network/network-context.service';
+import { getContractAt } from '../../modules/web3/contract';
+import { prisma } from '../../prisma/prisma-client';
+import { Multicaller } from '../web3/multicaller';
+import gaugeControllerAbi from './abi/gaugeController.json';
+export class VotingListService {
+    /**
+     * This query illustrates the pool related data that we need for each gauge
+     * Ideally, we don't want to denormalize but how can we join validGauges with getPoolsForVotingList in an efficient way?
+     */
+    public async getPoolsForVotingList(poolIds: string[]): Promise<any> {
+        let pools = await prisma.prismaPool.findMany({
+            where: { id: { in: poolIds } },
+            select: {
+                id: true,
+                address: true,
+                type: true, // TODO: Types are different than the original subgraph ones cause they are changed in mapSubgraphPoolTypeToPoolType (do we add a new column to prismaPool to store the original type?)
+                symbol: true,
+                tokens: {
+                    select: {
+                        address: true,
+                        dynamicData: {
+                            select: {
+                                weight: true,
+                            },
+                        },
+                        token: {
+                            select: {
+                                symbol: true,
+                            },
+                        },
+                    },
+                },
+            },
+        });
+
+        // Remove BPT
+        pools = pools.map((pool) => {
+            pool.tokens = pool.tokens.filter((t) => t.address !== pool?.address);
+            return pool;
+        });
+        // console.log('pools!', pools);
+        return pools;
+    }
+}
+
+export const votingListService = new VotingListService();
+
+/**
+ * Fetches the relative weight of all gauges using the GaugeController Contract following this steps:
+ * 1) onchain call to know the total number of gauges
+ * 2) multicall (in chunks of 100) to get the list of gauge addresses
+ * 3) multicall (in chunks of 100) to get the gauge_relative_weight for each gauge address
+ */
+export async function fetchGaugeRelativeWeights(): Promise<Record<string, string>> {
```

Comment:
My proposal would be:

- Add a schema into prisma that holds the rootGauges with an ID, isKilled, relativeWeightCap and a relation to PrismaStakingGauge (will need to add an optional reference there too)
- For the sync the process would be:
1) get all gauge addresses like you did below here
2) for each gauge address you need to get the chain (controller.gauge_types(addr))
3) for each gauge you need to get `is_killed` and `getRelativeWeightCap`
4) if it's Mainnet, the gauge address will be the liquidity gauge of the pool -> associate to the PrismaStakingGauge (of not found, special manual pool like veBAL)
5) if it's an L2, you need to query `getRecipient` as well to find the liquidity gauge address -> associate to the PrismaStakingGauge (of not found, special manual pool like veBAL)

Now the special manual gauges I'm not yet sure how to handle... Probably we need to keep track of all gauges that we couldn't associate and make sure that we have them as manual entries, otherwise we need to fire an alarm or an exception. 

Don't yet know how to best integrate the manual gauges.

Reply:
Why storing isKilled again if we already have it in the linked `PrismaStakingGauge`?

The important thing here is that we want to discard gauges which: 
isKilled = TRUE
relativeWeight = 0.0

Reply:
I'm not sure about the process, if I kill a root gauge, is the child gauge automatically killed as well? And vice versa?

Can you elaborate what `relativeWeight` is and why you need to check for that in addition to `isKilled`?

I would stick to the rootGauge name, as this is exactly what they are called in the contracts. Also, we would add all rootGauges into the DB, not just the ones that we can vote on.